pyscript.core/src/stdlib/pyscript.py

```diff
@@ -9,17 +9,19 @@
     from _pyscript.display import display
     from _pyscript import sync
 else:
-    # in MAIN both PyWorker and a runtime currentScript.id exist
-    # so these are both exposed and the display, if imported,
-    # will point at the right script as default target
-    PyWorker = window._pyscript.PyWorker
+    # in MAIN both PyWorker and current element target exist
+    # so these are both exposed and the display will use,
+    # if not specified otherwise, such current element target
+    import _pyscript_js
+    from _pyscript.display import display as py_display
 
-    def __getattr__(attribute_name):
-        if attribute_name == "display":
-            from _pyscript.display import display
+    PyWorker = _pyscript_js.PyWorker
 
-            id = window._pyscript.id
-            return lambda *values, target=id, append=True: display(
-                *values, target=target, append=append
-            )
-        raise AttributeError(f"'{__name__}' has no attribute '{attribute_name}'")
+    def current_target():
+        return _pyscript_js.target
```

Comment:
It might be good to define this also in a worker but make it raise a runtime error.

Reply:
it works as expected from a worker now, thanks!